### ISSUE TITLE
Enhancements and Improvements

### ISSUE THREAD

**Issue by mmatl** [mmatl/pyrender#8]
Making this issue to track potential improvements to pyrender:

- A deferred-rendering pipeline for improved performance with lots of meshes.
- An implementation of specular-glossiness materials in the shader.
- Cube maps for scene illumination.
- Shadow maps for point lighting.
- General performance improvements.

**Comment by bhack** [comment]
It is a little bit of topic but related to the python ecosystem. Too bad that [pyfilament](https://github.com/artometa/pyfilament) pybind11 binding was quite dead.
Filament has a quite interesting roadmap but currently it is not accessible from python and not so much progress was done for C wrapping (https://github.com/google/filament/issues/139).

**Comment by mmatl** [comment]
Yeah, I think an ideal version of this would be like a high-level python implementation of the glTF 2.0 standard with bindings down into filament or something similar. Maybe I'll try that eventually.

**Comment by bhack** [comment]
There was some activity on glTF 2.0 in python with pygltflib at https://gitlab.com/dodgyville/pygltflib but I don't know if it is still maintained now.
